collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -559,7 +562,7 @@ public void onClick(View v) {
                     if (folderIdStack.empty()) {
                         parentId = ROOT_KEY;
                     } else {
-                        parentId = folderIdStack.pop();
+                        parentId = folderIdStack.peek();
```

Comment:
What is it for? I think it's not related to this issue.

Reply:
This fixes the back button behaviour, before this change if we clicked on some folder and returned back to parent folder, the update messages on the forms were gone.

Reply:
Though i wonder why navigating up and down the hierarchy of folders was fine.